stream_executor/platform/mutex.h doesn't compile under C++14

Please go to Stack Overflow for help and support:

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Mac OS X with clang/llvm 5.0

- **TensorFlow installed from (source or binary)**:
source
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
0.4.5

### Describe the problem
There's an ifdef in mutex.h that uses shared_timed_mutex when compiled as C++14 and up. The file doesn't compile because C++14 requires using condition_variable_any rather than condition_variable with that kind of mutex.

### Source code / logs
```
In file included from external/org_tensorflow/tensorflow/stream_executor/platform/mutex.h:25:
external/org_tensorflow/tensorflow/stream_executor/platform/default/mutex.h:86:26: error: no matching member function for call to 'wait_for'
  std::cv_status s = cv->wait_for(*mu, std::chrono::milliseconds(ms));
                     ~~~~^~~~~~~~
external/org_chromium_clang_mac/include/c++/v1/__mutex_base:404:21: note: candidate function not viable: no known conversion from 'perftools::gputools::mutex_lock' to 'unique_lock<std::__1::mutex> &' for 1st argument
condition_variable::wait_for(unique_lock<mutex>& __lk,
                    ^
external/org_chromium_clang_mac/include/c++/v1/__mutex_base:426:21: note: candidate function template not viable: requires 3 arguments, but 2 were provided
condition_variable::wait_for(unique_lock<mutex>& __lk,
```

--- a/tensorflow/stream_executor/host/host_stream.h
+++ b/tensorflow/stream_executor/host/host_stream.h
@@ -48,7 +48,7 @@ class HostStream : public internal::StreamInterface {
 
   mutex mu_;
   int pending_tasks_ GUARDED_BY(mu_) = 0;
-  condition_variable completion_condition_;
+  ConditionVariableForMutex completion_condition_;
 };
 
 }  // namespace host

--- a/tensorflow/stream_executor/platform/default/mutex.h
+++ b/tensorflow/stream_executor/platform/default/mutex.h
@@ -42,8 +42,10 @@ enum ConditionResult { kCond_Timeout, kCond_MaybeNotified };
 
 #ifdef STREAM_EXECUTOR_USE_SHARED_MUTEX
 typedef std::shared_timed_mutex BaseMutex;
+typedef std::condition_variable_any ConditionVariableForMutex;
 #else
 typedef std::mutex BaseMutex;
+typedef std::condition_variable ConditionVariableForMutex;
 #endif
 
 // A class that wraps around the std::mutex implementation, only adding an
@@ -82,7 +84,7 @@ typedef mutex_lock shared_lock;
 using std::condition_variable;
 
 inline ConditionResult WaitForMilliseconds(mutex_lock* mu,
-                                           condition_variable* cv, int64 ms) {
+                                           ConditionVariableForMutex* cv, int64 ms) {
   std::cv_status s = cv->wait_for(*mu, std::chrono::milliseconds(ms));
   return (s == std::cv_status::timeout) ? kCond_Timeout : kCond_MaybeNotified;
 }

